gateway/src/inference/providers/openai.rs

```diff
@@ -120,6 +120,10 @@ impl InferenceProvider for OpenAIProvider {
         };
         Ok((chunk, stream))
     }
+
+    fn has_credentials(&self) -> bool {
+        self.api_key.is_some()
+    }
```

Comment:
If you're using an unauthenticated OpenAI-compatible endpoint, this will break.

We should check ~ "API base is None or API key is Some"